pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/SegmentCommitterFactory.java

```diff
@@ -29,18 +33,35 @@
 public class SegmentCommitterFactory {
   private static Logger LOGGER;
   private final ServerSegmentCompletionProtocolHandler _protocolHandler;
+  private final TableConfig _tableConfig;
+  private final ServerMetrics _serverMetrics;
+  private final IndexLoadingConfig _indexLoadingConfig;
 
-  public SegmentCommitterFactory(Logger segmentLogger, ServerSegmentCompletionProtocolHandler protocolHandler) {
+  public SegmentCommitterFactory(Logger segmentLogger, ServerSegmentCompletionProtocolHandler protocolHandler,
+      TableConfig tableConfig, IndexLoadingConfig indexLoadingConfig, ServerMetrics serverMetrics) {
     LOGGER = segmentLogger;
     _protocolHandler = protocolHandler;
+    _tableConfig = tableConfig;
+    _indexLoadingConfig = indexLoadingConfig;
+    _serverMetrics = serverMetrics;
   }
 
-  public SegmentCommitter createSplitSegmentCommitter(SegmentCompletionProtocol.Request.Params params,
-      SegmentUploader segmentUploader) {
-    return new SplitSegmentCommitter(LOGGER, _protocolHandler, params, segmentUploader);
-  }
+  public SegmentCommitter createSegmentCommitter(boolean isSplitCommit, SegmentCompletionProtocol.Request.Params params,
+      String controllerVipUrl) throws URISyntaxException {
+    if (!isSplitCommit) {
+      return new DefaultSegmentCommitter(LOGGER, _protocolHandler, params);
+    }
+    SegmentUploader segmentUploader;
+    if (_tableConfig.getValidationConfig().getPeerSegmentDownloadScheme() != null) {
```

Comment:
Can you add a TODO here? It looks odd that we upload to PinotFS if peer download scheme is set. 
We should have an independent config of where to upload segments.  I am not sure if this config should be:
(a) A table config (b) A server level config (c) Upload to whatever the controller asks us to do (so then becomes a cluster or controller level config).

I like (c) ,and definitely do not want (a) for this.

I would like to hear from you and @npawar and others if they have an opinion.

For now, this is ok since you need to move along with your implementation.

Reply:
done.

Reply:
Yes. The current way is not done cleanly. I will think more about this and fix it in a follow up discussion and PR. 

What we talk about here is how to construct different types of split committers as well as segment loaders.  server level config is a viable option. For the controller response option, the downside is that it will affect all servers behavior in a cluster.